programs/protocol-contracts-solana/src/lib.rs

```diff
@@ -106,43 +115,31 @@ pub mod gateway {
         Ok(())
     }
 
+    // deposit SOL into this program and the `receiver` on ZetaChain zEVM
+    // will get corresponding ZRC20 credit. The `receiver` should be a contract
+    // on zEVM and the `message` will be used as input data for the contract call.
+    // The `receiver` contract on zEVM will get the SOL ZRC20 credit and receive the `message`.
     pub fn deposit_and_call(
         ctx: Context<Deposit>,
         amount: u64,
         receiver: [u8; 20],
         message: Vec<u8>,
     ) -> Result<()> {
         require!(message.len() <= 512, Errors::MemoLengthExceeded);
-
-        let pda = &mut ctx.accounts.pda;
-        require!(!pda.deposit_paused, Errors::DepositPaused);
-
-        let cpi_context = CpiContext::new(
-            ctx.accounts.system_program.to_account_info(),
-            system_program::Transfer {
-                from: ctx.accounts.signer.to_account_info().clone(),
-                to: ctx.accounts.pda.to_account_info().clone(),
-            },
-        );
-        system_program::transfer(cpi_context, amount)?;
-        msg!(
-            "{:?} deposits {:?} lamports to PDA and call contract {:?} with message {:?}",
-            ctx.accounts.signer.key(),
-            amount,
-            receiver,
-            message,
-        );
-
+        deposit(ctx, amount, receiver)?;
```

Comment:
How/where do we emit the message so zEVM receives it?

Reply:
In the function parameter